packages/components/src/internal/components/search/actions.ts

```diff
@@ -145,3 +159,119 @@ export function getProcessedSearchHits(
               })
         : undefined;
 }
+
+export function getFinderSampleTypeNames(containerFilter: Query.ContainerFilter = undefined): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+        Query.executeSql({
+            // Retrieves all sample types for the given container filter, whether there are samples in them or not.
+            sql: 'SELECT Name FROM SampleSets',
+            containerFilter,
+            sort: 'Name',
+            schemaName: SCHEMAS.EXP_TABLES.SAMPLE_SETS.getSchema(),
+            success: data => {
+                resolve(data.rows.map(row => row['Name']));
+            },
+            failure: reason => {
+                console.error("Problem retrieving filtered sample types", reason);
+                reject("There was a problem retrieving the filtered sample types. " + resolveErrorMessage(reason));
+            }
+        });
+    });
+}
+
+const SAMPLE_FINDER_VIEW_NAME = "Sample Finder";
```

Comment:
are you not using `~~` in the view name because it works to just mark it has "hidden"?